BridgeSDK/BridgeAPI/BridgeAPITypes/SBBDateTimeRangeResourceList.m

```diff
@@ -55,17 +55,29 @@ - (void)reconcileWithDictionaryRepresentation:(NSDictionary *)dictionary objectM
     // as well, since that most likely means there was some kind of problem that required cleanup on the
     // server and the missing ones are no longer canonically valid. We'll look at the startTime and endTime
     // and remove our cached items in that date range (inclusive on the lower bound, exclusive on the upper),
-    // then add in the newly-retrieved ones, and sort by scheduledOn before adding back to
-    // the cached object's items[].
+    // then add in the newly-retrieved ones, and sort by date(s) before setting back as the cached object's items[].
     
-    NSString *comparisonKey = NSStringFromSelector(@selector(scheduledOn));
-    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K < %@ OR %K >= %@",
-                              comparisonKey, self.startTime,
-                              comparisonKey, self.endTime];
+    NSString *scheduledOnKey = NSStringFromSelector(@selector(scheduledOn));
+    NSString *expiresOnKey = NSStringFromSelector(@selector(expiresOn));
+    
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"(%K < %@ && %K != nil && %K < %@) OR %K >= %@",
+                              scheduledOnKey, self.startTime,
+                              expiresOnKey,
+                              expiresOnKey, self.startTime,
+                              scheduledOnKey, self.endTime];
```

Comment:
ok, as it turns out, this logic is actually correct for matching what the server returns for a given date range (and it needs to do that precisely here, for this to work).